pallets/multisig/src/lib.rs

```diff
@@ -600,14 +602,16 @@ decl_module! {
             ensure!(<MultiSigSignsRequired<T>>::contains_key(&multi_sig), Error::<T>::NoSuchMultisig);
             let sender_key = AccountKey::try_from(sender.encode())?;
             let sender_did = Context::current_identity_or::<Identity<T>>(&sender_key)?;
-            ensure!(
-                <MultiSigCreator<T>>::get(&multi_sig) == sender_did,
-                Error::<T>::IdentityNotCreator
-            );
+            let ms_key = AccountKey::try_from(multi_sig.encode())?;
+            if let Some(ms_identity) = <Identity<T>>::get_identity(&ms_key) {
```

Comment:
If possible could you please create an internal function to check the identity validation. It will increase the readability of the code.